Upgrade Documentation (#139)
cap-js/notifications: README.md

  - [Custom Notification Types Path](#custom-notification-types-path)
  - [Custom Notification Type Prefix](#custom-notification-type-prefix)
  - [Custom Destination Name](#custom-destination-name)
  - [Authentication Identifier](#authentication-identifier)
  - [Default Email Delivery](#default-email-delivery)
  - [Outbox Behavior](#outbox-behavior)
  - [Debug Logging](#debug-logging)
  - [Low-level Notifications API](#low-level-notifications-api)
- [Contributing](#contributing)
- [Code of Conduct](#code-of-conduct)
- [Licensing](#licensing)

## Setup

**Requirements:** Node.js >= 20, `@sap/cds` >= 8.

To enable notifications, simply add this self-configuring plugin package to your project:

```sh
npm add @cap-js/notifications
```

In this guide, we use the [Bookshop reference sample app](https://github.com/capire/bookshop) as the basis for publishing notifications.

## Getting Started

This section uses the Bookshop sample included in this repository. Everything is already wired up, you just need to run it, trigger a notification, and then make a change to see how the plugin responds.

### 1. Run the sample app

```sh
cd tests/bookshop
npm install
cds watch
```

> The server starts at `http://localhost:4004`. In development mode the plugin is active and notifications are printed to the console, no BTP account needed yet.

Reviewer: ```suggestion
The server starts at [http://localhost:4004](http://localhost:4004). During local development, the plugin prints out the notifications to the console and therefore no BTP account is needed yet.
```

```

`srv/book-ordered-email.html`:
```html
<h1>{i18n>BOOK_ORDERED_TITLE}</h1>
<p>Hi {{buyer}}, your order for <b>{{title}}</b> has been placed.</p>
```

#### Notification priority (Option A only)

**Static priority** - all notifications of a type share a fixed priority:

```cds
@notification.priority: #HIGH
event BookOrdered { ... }
```

Allowed values: `LOW`, `NEUTRAL` (default), `MEDIUM`, `HIGH`.

**Dynamic priority** - priority is computed at emit time from event data using a CDS ternary expression. The expression is evaluated against the database at runtime:

```cds
@notification.priority: (quantity > 5 ? #HIGH : #LOW)
event BookOrdered {
  quantity : Integer;
  title    : String;
}
```

More complex expressions using CDS functions are also supported:

```cds
@notification.priority: (days_between(orderDate, deliveryDate) > 7 ? #HIGH : #LOW)
event LateDelivery {
  orderDate    : Date;
  deliveryDate : Date;
}
```

#### Build integration

Running `cds build` also processes `@notification`-annotated events and writes a merged `notification-types.json` to the build output, combining types from CDS annotations with any types from the JSON file.

### Option B: JSON file

> As an alternative (or in addition) to CDS annotations, define types statically in `srv/notification-types.json` (this is the default expected location, see [Custom Notification Types Path](#custom-notification-types-path) to use a different path):

Reviewer: ```suggestion
As an alternative to CDS annotations, you can define types statically in `srv/notification-types.json` or a custom path (see [Custom Notification Types Path](#custom-notification-types-path) to use a different path):
```

## Licensing

> Copyright 2023–2026 SAP SE or an SAP affiliate company and contributors. Please see our [LICENSE](LICENSE) for copyright and license information. Detailed information including third-party components and their licensing/copyright information is available [via the REUSE tool](https://api.reuse.software/info/github.com/cap-js/notifications).

Reviewer: ```suggestion
Copyright 2026 SAP SE or an SAP affiliate company and contributors. Please see our [LICENSE](LICENSE) for copyright and license information. Detailed information including third-party components and their licensing/copyright information is available [via the REUSE tool](https://api.reuse.software/info/github.com/cap-js/notifications).
```
